From 1f881a48e173968cbd83892808b7992bd4d765ab Mon Sep 17 00:00:00 2001
Message-Id: <1f881a48e173968cbd83892808b7992bd4d765ab.1228954349.git.christian@jaeger.mine.nu>
In-Reply-To: <030d49bbacadc03ef604478be7d0582bd2703cf7.1228954349.git.christian@jaeger.mine.nu>
References: <030d49bbacadc03ef604478be7d0582bd2703cf7.1228954349.git.christian@jaeger.mine.nu>
From: Christian Jaeger <christian@jaeger.mine.nu>
Date: Wed, 10 Dec 2008 23:24:40 +0100
Subject: [PATCH] Need to still 'continue' (meaning, stop looping) after recursion (?)

I'm not sure why, but otherwise, we get this:

++ for node in '$nodes'
++ '[' 2 -eq 2 ']'
++ '[' -z '' ']'
++ opts=lvm=main-root
++ get_device_opts main-root-real lvm=main-root
++ local target source link extraopts rootopts opt
++ target=main-root-real
++ extraopts=lvm=main-root
++ KEYSCRIPT=
++ OPTIONS=
++ '[' -z main-root-real ']'
+++ grep '^main-root-real' /etc/crypttab
+++ head -1
+++ sed 's/[[:space:]]\+/ /g'
++ opt=
+++ echo
+++ cut -d ' ' -f2
++ source=
+++ echo
+++ cut -d ' ' -f3
++ key=
+++ echo
+++ cut -d ' ' -f4-
++ rootopts=
++ '[' -z '' ']'
++ echo 'cryptsetup: WARNING: invalid line in /etc/crypttab - '
cryptsetup: WARNING: invalid line in /etc/crypttab -
++ return 1
++ continue

Signed-off-by: Christian Jaeger <christian@jaeger.mine.nu>
---
 debian/initramfs/cryptroot-hook |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/debian/initramfs/cryptroot-hook b/debian/initramfs/cryptroot-hook
index 096c5bb..f8b197e 100644
--- a/debian/initramfs/cryptroot-hook
+++ b/debian/initramfs/cryptroot-hook
@@ -143,7 +143,7 @@ get_lvm_deps() {
 		fi
 		if [ "$(dmsetup table $depnode 2> /dev/null | cut -d' ' -f3)" != "crypt" ]; then
 			get_lvm_deps $depnode  # no quoting here as above there isn't either, and it contains appended whitespace
-			# continue  or not?
+			continue
 		fi
 		echo "$depnode"
 	done
-- 
1.6.0.5

